crates/vizia_core/src/views/textbox.rs

```diff
@@ -131,8 +107,12 @@ impl TextboxData {
             buf.action(match movement {
                 Movement::Grapheme(Direction::Upstream) => Action::Previous,
                 Movement::Grapheme(Direction::Downstream) => Action::Next,
+                Movement::Grapheme(Direction::Left) => Action::Left,
+                Movement::Grapheme(Direction::Right) => Action::Right,
```

Comment:
How do these differ from upstream and downstream above? And do we  need both?

Reply:
We might need both if, for example, the accessibility system has some input for "next character". We need Left and Right because they correspond to the actual intention of the user when they press the left or right arrow keys, and this corresponds to either upstream or downstream depending on whether the text is rtl or not.